crates/bevy_render/src/color/mod.rs

```diff
@@ -253,7 +253,7 @@ impl Color {
     /// ```
     ///
     pub fn hex<T: AsRef<str>>(hex: T) -> Result<Color, HexColorError> {
-        let hex = hex.as_ref();
+        let hex = hex.as_ref().replace('#', "");
```

Comment:
I don't think it's a wise idea to allocate a new `String` here like `replace` does. I would instead try to strip a leading `#`, if possible.

```suggestion
        let hex = hex.as_ref();
        let hex = hex.strip_prefix("#").unwrap_or(hex);
```

Not sure if this needs changes to `HexColorError` since the index it contains may now be off by 1 if there was a leading `#`.